### DISCUSSION TITLE
Clarification about generate_pseudo_pop() parameter

### DISCUSSION THREAD

**Discussion by m-qin** [NSAPH-Software/CausalGPS#141] [Q&A] [answered]
I’m wondering, in `generate_pseudo_pop()`, what is the difference between writing something like `params = list(xgb_nrounds = 30)` vs. `params = list(xgb_nrounds = c(10, 20, 30))`? What happens if we enter this `nrounds` parameter as vector versus entering `nrounds` as a single number? Thanks!

**Comment by Naeemkh** [comment, marked as answer]
If you enter `xgb_nrounds` (or any other parameter) as a vector and set `max_attempt` to any integer more than 1, the function internally will search among those parameters (up to max_attempt) to find the best combination for the covariate balance. 
For reproducible results, you can use the value you want to use and set `max_attempt = 1`.
